File: dotnet/Entites/Plan.cs
namespace dotnet.Entites
{
    public class Plan : BaseAuditableEntity
    {
        public Guid Id { get; set; }
        public Guid OwnerId { get; set; }
        public string Name { get; set; } = string.Empty;
        public string CoverImageUrl { get; set; } = string.Empty;
        public DateTimeOffset StartTime { get; set; }
        public DateTimeOffset EndTime { get; set; }
        public double Budget { get; set; }
        public string CurrencyCode { get; set; } = "USD";

        public User Owner { get; set; }
        public ICollection<ItineraryDay> ItineraryDays { get; set; } = new List<ItineraryDay>();
        public ICollection<ExpenseItem> ExpenseItems { get; set; } = new List<ExpenseItem>();
        public ICollection<PackingList> PackingLists { get; set; } = new List<PackingList>();
        public ICollection<Note> Notes { get; set; } = new List<Note>();
        public ICollection<Participant> Participants { get; set; } = new List<Participant>();
    }
}

File: react/types/plan.ts
import { ExpenseItem } from "./budget";
import { ItineraryDay } from "./itineraryDay";
import { Note } from "./note";
import { PackingList } from "./packingList";
import { Collaborator } from "./participant";

export type Plan = {
  id: string;
  ownerId: string;
  name: string;
  coverImageUrl: string | undefined;
  startTime: Date;
  endTime: Date;
  budget: number;
  currencyCode: string;

  // Owner info
  ownerName?: string;
  ownerUsername?: string;
  ownerAvatarUrl?: string;

  collaboratorId?: string;
  notes?: Note[];
  packingLists?: PackingList[];
  itineraryDays?: ItineraryDay[];
  expenseItems?: ExpenseItem[];
  collaborators?: Collaborator[];
  lastSyncGoogleCalendarAt?: string;
  createdAt?: string;
};
